### PR TITLE
Add warnings while upgrading RN with `react-native upgrade`

### PR DESCRIPTION
**Pull request by satya164**

### CODE REVIEW

File: local-cli/upgrade/upgrade.js [hunk 1]
```diff
@@ -9,16 +9,59 @@
...
+      );
+    } else {
+      const installed = JSON.parse(fs.readFileSync('node_modules/react-native/package.json', 'utf8'));
+
+      if (semver.satisfies(installed.version, version)) {
+        const v = version.replace(/^(~|\^|=)/, '');
+
+        console.log(
+          'Upgrading project to react-native v' + v + '\n' +
+          'Be sure to read the realease notes and breaking changes:\n' +
```
Comment (ide) [inline local-cli/upgrade/upgrade.js]: spelling: "release"

**Comment by satya164** [comment]
@ide Fixed.

**Comment by satya164** [comment]
cc @mkonicek

**Comment by satya164** [comment]
ping @mkonicek

File: local-cli/upgrade/upgrade.js [hunk 2]
```diff
@@ -9,16 +9,66 @@
...
+          console.log(
+            chalk.yellow(
+              'A valid version number for \'react-native\' is not specified your \'package.json\' file.'
+            )
+          );
+        }
+      } else {
+        console.warn(
+          chalk.yellow(
+            'react-native version in \'package.json\' doesn\'t match the installed version in \'node_modules\'.\n'
```
Comment (foghina) [inline local-cli/upgrade/upgrade.js]: Maybe add a note about running `npm install`?

**Comment by satya164** [comment]
@foghina Done.

**Comment by foghina** [comment]
It would be nice to also warn and bail out if the template already is at the current version. This is because people might think `upgrade` will also update their `react-native` dependency, but it doesn't. That's a bit of a trickier problem though so no need to do it in this PR.

**Comment by foghina** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**